test/tree.c

```diff
@@ -235,19 +235,19 @@ static void test_lookup_2(void)
 
 	c2 = nvme_lookup_ctrl(s, "tcp", "192.168.1.1", "192.168.1.20",
 			      "eth0", NULL, NULL);
-	assert(c1 == c2);
+	assert(c1 != c2);
```

Comment:
The changes need some more documentation why this behavior change here is correct. 

As far I understood @hreinecke and also @martin-gpy  the above test case is what expected to happen until now.  But with the new lookup implementation, this isn't the case anymore.

Reply:
Yep! That is exactly what I'm trying to solve. When a candidate wants a specific host-traddr and/or host-iface we need to make sure we match perfectly and not just return any controller that is "close enough".